source/dem/particle_particle_broad_search.cc

```diff
@@ -156,41 +114,47 @@ ParticleParticleBroadSearch<dim>::find_particle_particle_contact_pairs(
                 particles_in_neighbor_cell =
                   particle_handler.particles_in_cell(*cell_neighbor_iterator);
 
-              // Capturing particle pairs, the first particle (local) in the
-              // main cell and the second particle (ghost) in the neighbor
-              // cells
-              for (typename Particles::ParticleHandler<
-                     dim>::particle_iterator_range::iterator
-                     particles_in_main_cell_iterator =
-                       particles_in_main_cell.begin();
-                   particles_in_main_cell_iterator !=
-                   particles_in_main_cell.end();
-                   ++particles_in_main_cell_iterator)
+              // Capturing particle pairs, the first particle (local) in
+              // the main cell and the second particle (ghost) in the
+              // neighbor cells
+              for (auto particle_in_main_cell = particles_in_main_cell.begin();
+                   particle_in_main_cell != particles_in_main_cell.end();
+                   ++particle_in_main_cell)
                 {
-                  types::particle_index particle_one_id =
-                    particles_in_main_cell_iterator->get_id();
-
-                  std::vector<types::particle_index>
-                    &particle_candidate_container =
-                      ghost_contact_pair_candidates[particle_one_id];
-                  particle_candidate_container.reserve(40);
-
-                  for (typename Particles::ParticleHandler<
-                         dim>::particle_iterator_range::iterator
-                         particles_in_neighbor_cell_iterator =
-                           particles_in_neighbor_cell.begin();
-                       particles_in_neighbor_cell_iterator !=
-                       particles_in_neighbor_cell.end();
-                       ++particles_in_neighbor_cell_iterator)
-                    {
-                      particle_candidate_container.emplace_back(
-                        particles_in_neighbor_cell_iterator->get_id());
-                    }
+                  store_candidates(particles_in_neighbor_cell.begin(),
+                                   particles_in_neighbor_cell,
+                                   ghost_contact_pair_candidates
+                                     [particle_in_main_cell->get_id()]);
                 }
             }
         }
     }
 }
 
+template <int dim>
```

Comment:
This is a very very simple function. Can you move it to the .h and inline the function? This way you are 100% sure you won't loose performance ;)!